gateway/api_definition.go

```diff
@@ -817,16 +818,27 @@ func (a APIDefinitionLoader) getPathSpecs(apiVersionDef apidef.VersionInfo, conf
 	return combinedPath, len(whiteListPaths) > 0
 }
 
+var apiLangIDsRegex = regexp.MustCompile(`{([^}]*)}`)
+
 func (a APIDefinitionLoader) generateRegex(stringSpec string, newSpec *URLSpec, specType URLStatus, conf config.Config) {
-	apiLangIDsRegex := regexp.MustCompile(`{([^}]*)}`)
-	asRegexStr := apiLangIDsRegex.ReplaceAllString(stringSpec, `([^/]*)`)
+	// use mux to provide a regex path
+	asRegexStr, err := httputil.GetPathRegexp(stringSpec)
+	if err != nil {
+		// fallback to old behaviour in case of errors
+		// convert regexp params to a path match equivalent
+		asRegexStr = apiLangIDsRegex.ReplaceAllString(stringSpec, `([^/]*)`)
+	}
+
 	// Case insensitive match
 	if newSpec.IgnoreCase || conf.IgnoreEndpointCase {
```

Comment:
@jeffy-mathew we don't honor this for endpoint matching (granular access mw, rate limit per endpoint)

Reply:
not considered for now, resolving